pkg/registry/generic/registry/storage_factory.go

```diff
@@ -31,7 +31,24 @@ func StorageWithCacher(
 	resourcePrefix string,
 	scopeStrategy rest.NamespaceScopedStrategy,
 	newListFunc func() runtime.Object) storage.Interface {
-	return storage.NewCacher(
-		storageInterface, capacity, etcdstorage.APIObjectVersioner{},
-		objectType, resourcePrefix, scopeStrategy, newListFunc)
+
+	config := storage.CacherConfig{
+		CacheCapacity:  capacity,
+		Storage:        storageInterface,
+		Versioner:      etcdstorage.APIObjectVersioner{},
+		Type:           objectType,
+		ResourcePrefix: resourcePrefix,
```

Comment:
it is possible I feel. 

But can we try to do that in another pull request? It requires more thought. My general feeling was the cacher is a little bit complicated right now. I am trying to simplify it or at least make it easier to read.